src/Components/EventRegistration/index.jsx

```diff
@@ -0,0 +1,308 @@
+import { faCirclePlus, faXmark } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React, { useEffect, useState } from "react";
+import { Button, Form, Spinner } from "react-bootstrap";
+import { toast } from "react-toastify";
+import { createDoc, getDocById, getUsersInPECIdArray, updateDoc } from "services";
+
+const EventRegistration = ({
+  loader,
+  prelimLink,
+  setPrelimLink,
+  event,
+  onRegister,
+  userDets,
+  registeredTeamId = null,
+  alreadyRegistered = false,
+}) => {
+  const addTeamMemberSlot = () => {
+    if (teamMemberDetails.length < event.maxMembers) {
+      setErrorDetails((prevDets) => {
+        return [...prevDets, ""];
+      });
+      setTeamMemberDetails((prevDets) => {
+        return [...prevDets, ""];
+      });
+    }
+  };
+  const removeSlot = (ind) => {
+    if (teamMemberDetails.length > event.minMembers) {
+      setErrorDetails((prevDets) => {
+        const newArr = [...prevDets];
+        newArr.splice(ind, 1);
+        return newArr;
+      });
+      setTeamMemberDetails((prevDets) => {
+        const newArr = [...prevDets];
+        newArr.splice(ind, 1);
+        return newArr;
+      });
+    }
+  };
+  const [teamMemberDetails, setTeamMemberDetails] = useState(
+    Array.from({ length: event.minMembers }).fill("")
+  );
+  const [errorDetails, setErrorDetails] = useState(
+    Array.from({ length: event.minMembers }).fill("")
+  );
+ 
+  useEffect(() => {
+    if(!alreadyRegistered){
+      setTeamMemberDetails((prevDets) => {
+        const newArr = [...prevDets];
+        newArr[0] = userDets?.pecfestId;
+        return newArr;
+      });
+    }
+    else{
+      (async ()=>{
+        const teamData = await getDocById('event-teams', registeredTeamId);
+        console.log('teamData', teamData);
+        setTeamMemberDetails(teamData?.teamMembers);
+      })();
+    }
+   
+  }, []);
+  const validateUnique = () => {
+    const teamSet = new Set();
+    let finalValid = true;
+    teamMemberDetails.forEach((teamMember, ind) => {
+      if (!teamMember) {
+        return;
```

Comment:
Shouldn't this return false and error be added as teamMember shouldn't be empty?